src/chunk.c

```diff
@@ -2392,11 +2392,17 @@ ts_chunk_drop_process_materialization(Oid hypertable_relid,
 	ts_cache_release(hcache);
 }
 
+/* Continuous agg materialization hypertables can be dropped
+ * only if a user explicitly specifies the table name
+ * see discussion related to drop_chunks API:
+ * https://github.com/timescale/timescaledb/pull/1648
+ * API needs to be deprecated. this is the short term fix until then.
+ */
 List *
 ts_chunk_do_drop_chunks(Oid table_relid, Datum older_than_datum, Datum newer_than_datum,
 						Oid older_than_type, Oid newer_than_type, bool cascade,
 						CascadeToMaterializationOption cascades_to_materializations,
-						int32 log_level)
+						int32 log_level, bool user_supplied_table_name)
```

Comment:
It seems to me that this function could just be kept as is if we don't recurse to materialization tables in the user-visible call to `drop_chunks`, unless the user explicitly told us to:

```
drop_chunks(older_than => '1 week');   # <--- only drop chunks on user-created hypertables
drop_chunks(older_than => '1 week', cascade_to_materializations => true);   # <---  drop chunks on user-created hypertables and associated materializations
```